fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBNestedFieldQueryTest.java

```diff
@@ -1106,21 +1076,12 @@ void testNestedPrimaryKeyQuery() throws Exception {
 
         // Scan([[a, 2],[a, 2]])
         RecordQueryPlan plan = planner.plan(query);
-        if (planner instanceof RecordQueryPlanner) {
-            assertMatchesExactly(plan,
-                    scanPlan()
-                            .where(scanComparisons(range("[[a, 2],[a, 2]]"))));
-            assertEquals(1265534819, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
-            assertEquals(-1094245019, plan.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
-            assertEquals(-2039475834, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
-        } else {
-            assertMatchesExactly(plan,
-                    scanPlan()
-                            .where(scanComparisons(range("[[a, 2],[a, 2]]"))));
-            assertEquals(1265534819, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
-            assertEquals(1873510398, plan.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
-            assertEquals(928279583, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
-        }
+        assertMatchesExactly(plan,
+                scanPlan()
+                        .where(scanComparisons(range("[[a, 2],[a, 2]]"))));
+        assertEquals(1265534819, plan.planHash(PlanHashable.CURRENT_LEGACY));
+        assertEquals(767624337, plan.planHash(PlanHashable.CURRENT_FOR_CONTINUATION));
```

Comment:
Does this imply a plan hash change for this scan plan?

Reply:
There are some cases where there was an `if` around the planner in use that I collapsed. I don't think the plan hash is changed. It just shows up as a diff.